### PR TITLE
[Go] change vertexai plugin to new style

### PR DESCRIPTION
**Pull request by jba**
Same change as to googleai.
Add Init function, and Model and Embedder functions to retrieve them.

### CODE REVIEW

File: go/plugins/vertexai/vertexai.go [hunk 1]
```diff
@@ -16,15 +16,97 @@ package vertexai
...
 	"github.com/firebase/genkit/go/ai"
 	"github.com/firebase/genkit/go/plugins/internal/uri"
+	"google.golang.org/api/option"
 )
 
-func newClient(ctx context.Context, projectID, location string) (*genai.Client, error) {
-	return genai.NewClient(ctx, projectID, location)
+const provider = "google-genai"
+
+// Config configures the plugin.
```
Comment (randall77) [inline go/plugins/vertexai/vertexai.go:32]: Config provides configuration options for the Init function.

File: go/plugins/vertexai/vertexai.go [hunk 2]
```diff
@@ -16,15 +16,97 @@ package vertexai
...
+		return errors.New("need at least one model or embedder")
+	}
+
+	// Client for Gemini SDK.
+	gclient, err := genai.NewClient(ctx, cfg.ProjectID, cfg.Location)
+	if err != nil {
+		return err
+	}
+
+	// Client for prediction SDK, needed for embedders.
```
Comment (randall77) [inline go/plugins/vertexai/vertexai.go:69]: This code seems to alternate between model setup and embedder setup. Maybe we should do all one, then all of the other. Maybe in separate functions?

**Review by randall77 — APPROVED** [review]

File: go/plugins/vertexai/vertexai.go (hunk 2 above)
Comment (jba) [reply go/plugins/vertexai/vertexai.go]: Done.

File: go/plugins/vertexai/vertexai.go (hunk 1 above)
Comment (jba) [reply go/plugins/vertexai/vertexai.go]: Done.

**Pull request merged**